### PR TITLE
[Refactor] Refactor the image_* command flagging process

### PR DESCRIPTION
**Pull request by Zheaoli**
Checklist:

- [x] Create a file in `pkg/api/types/${cmd}_types.go`, and define the CommandOption for this command
- [x] Create some file in `pkg/cmd/${cmd}`, and move the command entry point in real into this package

Signed-off-by: Zheao.Li <me@manjusaka.me>

### CODE REVIEW

File: pkg/api/types/image_types.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	// #region overlaybd flags
+	// Overlaybd convert tar.gz layers to overlaybd layers
+	Overlaybd bool
+	// OverlayFsType filesystem type for overlaybd
+	OverlayFsType string
+	// OverlaydbDBStr database config string for overlaybd
+	OverlaydbDBStr string
+	// #endregion
+
+	// #region generic flags
```
Comment (djdongjin) [inline pkg/api/types/image_types.go:96]: nit: maybe we should put generic/common flags before flags specific to e.g., stargz/nydus/etc

File: pkg/api/types/image_types.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+	GOptions GlobalCommandOptions
+	// Platforms convert content for a specific platform
+	Platforms []string
+	// AllPlatforms convert content for all platforms
+	AllPlatforms bool
+	// GpgHomeDir the GPG homedir to use; by default gpg uses ~/.gnupg"
+	GpgHomeDir string
+	// GpgVersion the GPG version ("v1" or "v2"), default will make an educated guess
+	GpgVersion string
+	// Keys a secret key's filename and an optional password separated by colon; this option may be provided multiple times"\
```
Comment (djdongjin) [inline pkg/api/types/image_types.go:121]: remove `this option may be provided multiple times"\` (helpful as command usage but here we're using a `[]string`, so not need)

File: pkg/cmd/image/convert.go [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+	estargzconvert "github.com/containerd/stargz-snapshotter/nativeconverter/estargz"
+	estargzexternaltocconvert "github.com/containerd/stargz-snapshotter/nativeconverter/estargz/externaltoc"
+	zstdchunkedconvert "github.com/containerd/stargz-snapshotter/nativeconverter/zstdchunked"
+	"github.com/containerd/stargz-snapshotter/recorder"
+	"github.com/klauspost/compress/zstd"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/sirupsen/logrus"
+)
+
+func Convert(ctx context.Context, options types.ImageConvertCommandOptions, stdout io.Writer, srcRawRef, targetRawRef string) error {
```
Comment (djdongjin) [inline pkg/cmd/image/convert.go:49]: nit: maybe just personal preference, but I feel we should move `srcRawRef, targetRawRef string` as the 2nd/3rd args after `ctx`. It's the most relevant to `Convert` (same for other funcs)

File: pkg/cmd/image/push.go [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+		if options.Estargz {
+			layerConvert = eStargzConvertFunc()
+		}
+		c, err := ipfs.Push(ctx, client, ref, layerConvert, options.AllPlatforms, options.Platforms, options.IpfsEnsureImage, ipfsPath)
+		if err != nil {
+			logrus.WithError(err).Warnf("ipfs push failed")
+			return err
+		}
+		fmt.Fprintln(stdout, c)
+		return err
```
Comment (ktock) [inline pkg/cmd/image/push.go:84]: return nil

**Review by ktock — APPROVED** [review]

**Comment by AkihiroSuda** [comment]
@ktock You can merge

**Pull request merged**